app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -291,31 +302,182 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
+    }
+
+    /**
+     * @return boolean
+     */
+    protected function allowOutOfStockOrders()
+    {
+        return Mage::getStoreConfigFlag('payment/boltpay/allow_out_of_stock_orders');
+    }
+
+    /**
+     * @param Mage_Sales_Model_Quote $quote
+     *
+     * @return void
+     */
+    protected function validateProducts(Mage_Sales_Model_Quote $quote)
+    {
+        $outOfStockSKUs = $this->getOutOfStockSKUs($quote);
+        if ($outOfStockSKUs) {
+            $this->enableOutOfStockOrderToBeCreated();
+            $this->setOrderOnHoldFlag();
+
+            $errorMessage = Mage::helper('boltpay')->__("Product " .
+                join(", ", $outOfStockSKUs) .
+                (count($outOfStockSKUs) == 1 ? " is" : " are") .
+                " out of stock. ");
+
+            $this->appendOrderOnHoldMessage($errorMessage);
+        }
+
+        $disabledSKUs = $this->getDisabledSKUs($quote);
+        if ($disabledSKUs) {
+            $this->setOrderOnHoldFlag();
+
+            $errorMessage = Mage::helper('boltpay')->__("Product " .
+                join(", ", $disabledSKUs) .
+                (count($disabledSKUs) == 1 ? " is" : " are") .
+                " disabled. ");
+
+            $this->appendOrderOnHoldMessage($errorMessage);
+        }
+
+        if ($this->isOrderOnHoldFlagSet()) {
+            $invalidSKUs = array_unique(array_merge($outOfStockSKUs, $outOfStockSKUs));
+            $errorMessage = Mage::helper('boltpay')->__("Please review " .
+                (count($invalidSKUs) > 1 ? "them" : "it") .
+                " and un-hold the order. ");
+
+            $this->appendOrderOnHoldMessage($errorMessage);
+        }
+    }
+
+    /**
+     * @return void
+     */
+    protected function setOrderOnHoldFlag()
+    {
+        $this->isOrderOnHold = true;
+    }
+
+    /**
+     * @return boolean
+     */
+    protected function isOrderOnHoldFlagSet()
+    {
+        return $this->isOrderOnHold;
+    }
+
+    /**
+     * @param $message
      *
-     * @var Mage_Sales_Model_Quote $quote   The quote that defines the cart
+     * @return void
+     */
+    protected function appendOrderOnHoldMessage($message)
+    {
+        $this->orderOnHoldMessage .= $message;
+    }
+
+    /**
+     * @var Mage_Sales_Model_Quote $quote The quote that defines the cart
      *
-     * @return bool true if the store can accept an order for all items in the cart,
-     *              otherwise, false
+     * @return array
      */
-    public function storeHasAllCartItems($quote)
+    public function getOutOfStockSKUs(Mage_Sales_Model_Quote $quote)
     {
-        foreach ($quote->getAllItems() as $cartItem) {
-            if($cartItem->getHasChildren()) {
-                continue;
+        if($this->outOfStockSkus == null) {
+            $this->outOfStockSkus = array();
+
+            foreach($this->getCartProducts($quote) as $product) {
+                $stockInfo = Mage::getModel('cataloginventory/stock_item')->loadByProduct($product);
```

Comment:
Can you check is it work here?
I thought that we can get info directly from product like
`$product->getStockItem()`

It should load product Stock Item object from cache like here

`
class  Mage_Catalog_Model_Convert_Adapter_Product

    public function getStockItem()
    {
        return Mage::objects()->load(Mage::registry('Object_Cache_StockItem'));
    }
`